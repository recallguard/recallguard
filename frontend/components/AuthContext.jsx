import { createContext, useState, useEffect } from 'react';

export const AuthContext = createContext({ token: null, login: () => {}, logout: () => {} });

export function AuthProvider({ children }) {
  const [token, setToken] = useState(null);

  useEffect(() => {
    const t = localStorage.getItem('token');
    if (t) setToken(t);
  }, []);

  const login = (t) => {
    localStorage.setItem('token', t);
    setToken(t);
  };

  const logout = () => {
    localStorage.removeItem('token');
    setToken(null);
  };

  return (
    <AuthContext.Provider value={{ token, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
}
