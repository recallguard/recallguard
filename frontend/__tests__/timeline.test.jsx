import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';

jest.mock('next/router', () => ({ useRouter: () => ({ query: { id: '1' } }) }));

import RecallDetail from '../pages/recall/[id].jsx';
import { AuthContext } from '../components/AuthContext.jsx';

test('timeline snapshot', async () => {
  global.fetch = jest.fn().mockResolvedValue({
    json: async () => ({
      product: 'Widget',
      hazard: 'Fire',
      remedy_updates: [
        { time: '2024-01-01', text: 'Initial' },
        { time: '2024-02-01', text: 'Follow-up' }
      ]
    })
  });
  const { asFragment, findByText } = render(
    <AuthContext.Provider value={{ token: 't' }}>
      <RecallDetail />
    </AuthContext.Provider>
  );
  await findByText('Widget');
  expect(asFragment()).toMatchSnapshot();
});
