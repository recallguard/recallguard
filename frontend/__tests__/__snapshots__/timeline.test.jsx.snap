// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`timeline snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Widget
    </h2>
    <p>
      Fire
    </p>
    <div
      aria-label="timeline"
    >
      <div
        style="border-left: 2px solid rgb(204, 204, 204); margin-left: 1em; padding-left: 1em;"
      >
        <div>
          2024-01-01
        </div>
        <p>
          Initial
        </p>
      </div>
      <div
        style="border-left: 2px solid rgb(204, 204, 204); margin-left: 1em; padding-left: 1em;"
      >
        <div>
          2024-02-01
        </div>
        <p>
          Follow-up
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
